### ISSUE TITLE
(Flaky-test) PrometheusMetricsTest.testPerNamespaceStats

### ISSUE THREAD

**Issue by devinbost** [apache/pulsar#6352]
I'm getting this intermittent (infrequent) issue:

```
[ERROR] Tests run: 5, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 2.503 s <<< FAILURE! - in org.apache.pulsar.broker.stats.PrometheusMetricsTest
[ERROR] testPerNamespaceStats(org.apache.pulsar.broker.stats.PrometheusMetricsTest)  Time elapsed: 0.204 s  <<< FAILURE!
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
	at java.util.ArrayList.rangeCheck(ArrayList.java:657)
	at java.util.ArrayList.get(ArrayList.java:433)
	at org.apache.pulsar.broker.stats.PrometheusMetricsTest.testPerNamespaceStats(PrometheusMetricsTest.java:232)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```
(https://github.com/apache/pulsar/pull/6202/checks?check_run_id=451988656)

It may have been masked by other failing tests in PrometheusMetricsTest that I fixed with PR #6202. 

I've uploaded the test log output: [org.apache.pulsar.broker.stats.PrometheusMetricsTest-output.txt](https://github.com/apache/pulsar/files/4217989/org.apache.pulsar.broker.stats.PrometheusMetricsTest-output.txt)

**Comment by devinbost** [comment]
This is fixed in #6202

**Comment by tisonkun** [comment]
Closed as stale and no recent report. Please open a new issue if it's still relevant in maintained versions.
